JMJD6.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def seq_len():
+        pass
+
+    def seq_slices():
+        pass
+
+    def seq_indexes():
+        pass
+
+    def seq_to_string():
+        pass
+
+    def check_alphabet():
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    COMPLEMENTARITY_DICT = {"A": "T", "T": "A", "C": "G", "G": "C"}
+    alphabet = ["A", "T", "U", "G", "C"]
+
+    def __init__(self, seq):
+        self.seq = seq
+
+    def seq_len(self):
+        return len(self.seq)
+
+    def seq_slices(self, start, stop):
+        slice = self.seq[start:stop]
+        return slice
+
+    def seq_indexes(self, index):
+        return self.seq[index]
+
+    def seq_to_string(self):
+        return str(self.seq)
+
+    def check_alphabet(self):
+        if set(self.seq) <= set(self.alphabet):
+            return 'That`s a nucleic acid sequence'
+        else:
+            return 'That`s not a nucleic acid sequence'
+
+    def complement(self):
+        complement = ""
+        for nucl in self.seq:
+            complement += self.COMPLEMENTARITY_DICT.get(nucl)
+        return complement

```

Comment:
Меняется тип данных. Возвращается просто строка. Лучше переписать так:
```suggestion
        complement = ""
        for nucl in self.seq:
            complement += self.COMPLEMENTARITY_DICT.get(nucl)
        return type(self)(complement)
```